enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,600 @@
+upgrading with this unsupported configuration.
+
+If the `service.beta.kubernetes.io/aws-load-balancer-subnets` annotation was
+previously configured and the cluster is upgraded, the default value `[]` for
+`Subnets` will clear the annotation for existing IngressControllers, which
+would break cluster ingress. However, requiring the service to be deleted to effectuate
+a `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+value different from the current value of the annotation prevents this
+automatic removal of the service annotation on upgrade.
+
+**Note**: Cluster admins upgrading with an unmanaged subnet annotation don't
+need to delete the service to propagate the subnet values, instead they can follow the
+[Unmanaged Subnet Annotation Migration Workflow](#unmanaged-subnet-annotation-migration-workflow).
+
+##### Reason #3: The CCM Doesn't Reconcile NLB Subnets Updates after creation.
+
+The CCM doesn't reconcile `service.beta.kubernetes.io/aws-load-balancer-subnets`
+annotation updates to NLBs once it has been created. However, requiring a service
+deletion will effectuate these changes.
+
+See [CCM Doesn't Reconcile NLB Subnets Updates](#ccm-doesnt-reconcile-nlb-subnets-updates)
+for more details on why the CCM doesn't reconcile these updates.
+
+##### Auto Effectuation with auto-delete-load-balance Annotation
+
+In addition to deleting the IngressController explicitly, it is possible to use the existing
+`ingress.operator.openshift.io/auto-delete-load-balancer` annotation to instruct the Ingress Operator to automatically
+delete the service after a subnet update. While this annotation was initially introduced to enable automatic scope
+changes in [Ingress Mutable Publishing Scope](/enhancements/ingress/mutable-publishing-scope.md), we've chosen to extend
+its usage to support the `Subnets` fields. The auto-delete annotation is not intended for end-users to use directly, but
+instead for configuration management tooling.
+
+### Implementation Details/Notes/Constraints
+
+When an IngressController is created with
+`spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+specified, the Ingress Operator will set the
+`service.beta.kubernetes.io/aws-load-balancer-subnets` annotation
+on the LoadBalancer-type Service.
+
+However, if `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+is updated on an existing IngressController, the Ingress Operator will NOT
+update `service.beta.kubernetes.io/aws-load-balancer-subnets` until the
+cluster admin deletes the LoadBalancer-type service as instructed by the
+`LoadBalancerProgressing` status (see [Effectuating Subnet Updates](#effectuating-subnet-updates)).
+When the Ingress Operator recreates the LoadBalancer-type service, it will
+then configure `service.beta.kubernetes.io/aws-load-balancer-subnets` with
+the new `Subnets` value.
+
+The only exception is if the `ingress.operator.openshift.io/auto-delete-load-balancer`
```

Comment:
nit: this belongs with the previous paragraph.
```suggestion
the new `Subnets` value.  The only exception is if the 
`ingress.operator.openshift.io/auto-delete-load-balancer`
```

Reply:
Yep I think I went back and forth on this. Done.